cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1503,7 +1553,7 @@ void writer::impl::write(table_view const& table, std::vector<partition_info> co
   std::transform(partitions.begin(),
                  partitions.end(),
                  std::back_inserter(num_frag_in_part),
-                 [this, max_page_fragment_size](auto const& part) {
+                 [&, max_page_fragment_size](auto const& part) {
```

Comment:
why is the capture-default (&) needed here?

Reply:
That is to replace `this` which is not possible for a free function. Probably it is even not needed.